vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1564,10 +1566,16 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
             temp_rr_segs.push_back(rr_seg);
         }
 
+        t_unified_to_parallel_seg_index seg_index_map;
+        auto segment_inf_x_ = get_parallel_segs(temp_rr_segs, seg_index_map, X_AXIS);
+        auto segment_inf_y_ = get_parallel_segs(temp_rr_segs, seg_index_map, Y_AXIS);
         alloc_and_load_rr_indexed_data(
             temp_rr_segs,
+            segment_inf_x_,
+            segment_inf_y_,
             *wire_to_rr_ipin_switch_,
-            base_cost_type_);
+            base_cost_type_,
+            lookahead_type_);
```

Comment:
So the lookahead is only used to determine how we calculate the ortho-cost indices which are only relevant to the CLASSIC lookahead (I think?). To get around this, I added a field to `e_lookahead_type` which serves as a placeholder for a situation like this and removed all other instances of a field in the serialize object.

Reply:
Why not just always set up the data?  If it is written out /read in with the rr-graph we should just set the data regardless of what lookahead is currently set.

Reply:
See latest comment. So we actually do setup the data regardless but if we're using the CLASSIC lookahead, we pay extra-overhead for rr_graph full traversal.

Reply:
How long does it take?  If it's not much (and one traversal shouldn't be much) I'd just compute it all the time.  Or do you have the caller always pass in to compute the extra data anyway (in the writer)?  Otherwise we seem to have a complicated order dependency on using the classic lookahead.

If you leave this, it should also be commented (this point should go in there to document why this is passed in and what it controls).

Reply:
Sure, I can make the change so the graph is always traversed. To clarify, the traversal is not O(V), but O(E). The writer does this traversal `find_ortho_cost_index` in the below code. 

```
if (lookahed_type == e_router_lookahead::CLASSIC) {
        ortho_costs = find_ortho_cost_index(segment_inf_x, segment_inf_y, X_AXIS);

    } else {
        std::vector<int> x_costs(segment_inf_x.size(), CHANX_COST_INDEX_START + segment_inf_x.size());
        std::vector<int> y_costs(segment_inf_y.size(), CHANX_COST_INDEX_START);

        std::move(x_costs.begin(), x_costs.end(), std::back_inserter(ortho_costs));
        std::move(y_costs.begin(), y_costs.end(), std::back_inserter(ortho_costs));
    }
```

Reply:
I'd time it on a big circuit and compare it to total runtime and rr-graph construction time.  If it isn't much we can just do the more complex calculation.  We can leave the alternative code path in case we get rid of the classic lookahead at some point.

Reply:
Sure sounds good.

Reply:
@vaughnbetz added results for titan_qor for calculating `ortho_cost_indeces ` with and without rr_graph traversal. Like you suspected, difference seems small: https://docs.google.com/spreadsheets/d/1yweCJASc03fyT9cKKC6AAdauPH0zbrDbAoEE_5HUV50/edit?usp=sharing